### PR TITLE
Remove some redundant abstractions

### PR DESCRIPTION
**Pull request by lukebjerring**
## Description
Not sure the exact problem, but mockgen is choking on these files.
Obviated the problem by just deleting the interfaces.

### CODE REVIEW

**Comment by Hexcles** [comment]
Could you paste the mockgen errors? I don't see mockgen errors on Travis, but only the build error caused by `unsafe`.

**Comment by lukebjerring** [comment]
Can't repro anymore, but as it was, there's nothing to post other than 

> Failed to load package

(mockgen doesn't print anything other than that), so not very informative. 

While not blocked on this change, it does simplify things somewhat, so I'll leave the PR as-is.

**Comment by Hexcles** [comment]
Yeah mockgen was broken at one point and I fixed it. I was wondering if you
got caught in some intermediate state.

File: api/query/cache/backfill/backfill.go [hunk 1]
```diff
@@ -52,31 +38,21 @@ const bytesPerRun = uint64(6.5e+7)
 
 var errNilIndex = errors.New("Index to backfill is nil")
 
-// NewDatastoreRunFetcher constructs a RunFetcher that loads runs from Datastore
+// GetDatastore constructs a RunFetcher that loads runs from Datastore
```
Comment (Hexcles) [inline api/query/cache/backfill/backfill.go]:
```suggestion
// GetDatastore constructs a shared.Datastore interface that loads runs from Datastore
```

**Comment by Hexcles** [comment]
Sorry for the delay. I pushed a few commits to fix some nits (including a typo that broke the build) myself.

**Pull request merged**